core/src/main/java/io/ddf/content/AMetaDataHandler.java

```diff
@@ -59,9 +59,19 @@ public long getNumRows() throws DDFException {
     return mNumRows;
   }
 
-  public void copyFactor(DDF oldDDF)  throws DDFException {
-
-  };
+  /**
+   * Transfer factor information from ddf to this DDF
+   * @param ddf
+   * @throws DDFException
+   */
+  public void copyFactor(DDF ddf)  throws DDFException {
+    for (Schema.Column col : ddf.getSchema().getColumns()) {
+      if (this.getDDF().getColumn(col.getName()) != null && col.getColumnClass() == Schema.ColumnClass.FACTOR) {
```

Comment:
Good that you check null here. I will also return an error message so user know what is going on.